haystack/schema.py

```diff
@@ -1317,7 +1319,14 @@ def _build_answer_metrics_df(
                 ].unique()
                 query_answers = answers[answers["multilabel_id"] == multilabel_id]
                 # consider only the answers within simulated_top_k_retriever documents
-                simulated_query_answers = query_answers[query_answers["document_id"].isin(top_k_document_ids)]
+
+                simulated_query_answers = query_answers[
+                    query_answers["document_ids"].apply(
+                        lambda document_ids, top_k_document_ids=top_k_document_ids: all(
+                            document_id in top_k_document_ids for document_id in document_ids
+                        )
+                    )
+                ]
```

Comment:
We discard all answers, that used any document that was not under the top_k document. This looks way more complex now, unfortunately.